Nos/Views/LikeButton.swift

```diff
@@ -65,18 +65,41 @@ struct LikeButton: View {
     
     var body: some View {
         Button {
-            tapped = true
             Task {
-                await action()
+                await buttonPressed()
             }
         } label: {
             buttonLabel
         }                             
-        .disabled(currentUserLikesNote || tapped)
+        .disabled(tapped)
```

Comment:
I think `tapped` now acquired a different meaning and two responsibilities.

1. It fills the like button immediately to give instant feedback while the like is being processed
2. It prevents the user from double tapping while the async operation is being done

Right now, the 2nd bullet is only prevented when liking but not when deleting the like. And the name of the bool (tapped) is confusing because it is used to display the button as "liked" state, not pressed.

As we have an additional operation, delete like, we need to maintain this tapped bool that fulfills the 2nd bullet and have a different way of fulfilling the first bullet. The first thing that pops to my head is having a different bool for that, maybe there is a simpler solution that avoid having two different bools.

Reply:
You're absolutely right. Time for a new `State` type (my solution to everything)! 😂

Reply:
Done!